lib/managers/connection_check_status_manager.dart

```diff
@@ -0,0 +1,50 @@
+import 'dart:async';
+
+import 'package:connectivity_plus/connectivity_plus.dart';
+
+import 'package:api/models/app/managers/connection_status_manager.dart';
+
+class ConnectionCheckStatusManager implements ConnectionStatusManager {
+  static final ConnectionCheckStatusManager _connectionStatusSingleton =
+      ConnectionCheckStatusManager._internal();
+  ConnectionCheckStatusManager._internal();
+
+  factory ConnectionCheckStatusManager() => _connectionStatusSingleton;
+
+  bool _hasConnection = false;
+
+  final StreamController _connectionChangeController =
+      StreamController<bool>.broadcast();
+
+  final Connectivity _connectivity = Connectivity();
+
+  @override
+  void activateConnectionManager() => _connectivity.onConnectivityChanged
+      .asBroadcastStream()
+      .listen(_getConnectionChange);
+
+  void _getConnectionChange(ConnectivityResult result) =>
+      hasInternetConnection();
+
+  @override
+  Stream get connectionChangeStream => _connectionChangeController.stream;
```

Comment:
Seems like it's not used in any viewModel or file. 
If not, delete the method here and in the abstract class